File: MSB-backend/src/main/java/entity/Facility.java
package entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "facility")
public class Facility {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "facility_id")
    private long facilityId;

    @Column(name = "facility_name")
    private String facilityName;

    @Enumerated(EnumType.STRING)
    @Column(name = "facility_type")
    private FacilityType facilityType;

    private String address;

    private String location;

    @Column(name = "phone_number")
    private String phoneNumber;

    private String email;

    private double ratings;

    private String amenities;

    @ManyToMany(mappedBy = "facilities")
    private Set<Event> events;

    @OneToMany(mappedBy = "facility")
    private Set<Booking> bookings;
}

File: MSB-frontend/src/app/facility-manager/facility-manager-create/facility.ts
export interface Facility {
  facilityId?: number;
  facilityName: string;
  facilityType: string;  
  address: string;
  location: string;
  phoneNumber: string;
  email: string;
  ratings: number;
  amenities: string;
  facilityManager:
    {
      userId: number;
      firstName: string;
      lastName: string;
    };

}
